src/app/pages/RewardPage/components/HistoryTable/index.tsx

```diff
@@ -1,85 +1,48 @@
-import axios from 'axios';
-import React, { useCallback, useEffect, useState } from 'react';
-
-import { Pagination } from 'app/components/Pagination';
-import { useAccount, useBlockSync } from 'app/hooks/useAccount';
-import { backendUrl, currentChainId } from 'utils/classifiers';
-
-import { RewardEvent, TableBody } from './TableBody';
+import React, { useCallback, useMemo, useState } from 'react';
+import { TableBody } from './TableBody';
 import { TableHeader } from './TableHeader';
-import { RewardTabType } from '../../types';
-
-const pageSize = 6;
-
-const getHistoryEndpoint = (activeTab: RewardTabType) => {
-  switch (activeTab) {
-    case RewardTabType.REWARD_SOV:
-      return 'rewardsNew';
-    case RewardTabType.LIQUID_SOV:
-      return 'liquidSov';
-    case RewardTabType.FEES_EARNED:
-      return 'feesEarned';
-    default:
-      return 'rewardsNew';
-  }
-};
+import { useGetUserRewardsEarnedHistory } from '../../hooks/useGetUserRewardsEarnedHistory';
+import { RewardPagination } from '../RewardPaginaton';
 
-interface IHistoryTableProps {
-  activeTab: RewardTabType;
-}
+const PAGE_SIZE = 5;
 
-export const HistoryTable: React.FC<IHistoryTableProps> = ({ activeTab }) => {
-  const account = useAccount();
-  const url = backendUrl[currentChainId];
-  const [history, setHistory] = useState<RewardEvent[]>([]);
-  const [loading, setLoading] = useState(false);
+export const HistoryTable: React.FC = () => {
   const [page, setPage] = useState(1);
-  const [total, setTotal] = useState(0);
-  const blockSync = useBlockSync();
-
-  const getHistory = useCallback(() => {
-    setLoading(true);
-    setHistory([]);
-
-    const historyEndpoint = getHistoryEndpoint(activeTab);
-
-    axios
-      .get(
-        `${url}/v1/event-history/${historyEndpoint}/${account}?page=${page}&pageSize=${pageSize}`,
-      )
-      .then(res => {
-        const { events, pagination } = res.data;
-        setHistory(events || []);
-        setTotal(pagination.totalPages * pageSize);
-        setLoading(false);
-      })
-      .catch(e => {
-        console.log(e);
-        setLoading(false);
-      });
-  }, [activeTab, url, account, page]);
+  const { data, loading } = useGetUserRewardsEarnedHistory(page, PAGE_SIZE);
+  const isHiddenPagination = useMemo(
+    () =>
+      data &&
+      data.rewardsEarnedHistoryItems.length < PAGE_SIZE &&
+      page === 1 &&
+      !loading,
+    [data, page, loading],
+  );
 
-  useEffect(() => {
-    if (account) {
-      getHistory();
+  const isDisabled = useMemo(() => {
```

Comment:
Change it to `  const isDisabled = useMemo(
    () => !data || (data && data.rewardsEarnedHistoryItems.length < PAGE_SIZE),
    [data],
  );`